src/piggui.rs

```diff
@@ -109,6 +109,7 @@ impl Application for Piggui {
                         let _ = self
                             .toast_handler
                             .update(ToastMessage::UnsavedChangesExitToast, &self.hardware_view);
+                        self.unsaved_changes = false;
```

Comment:
This is the line that fixes that bug, right?
You could add a "Fixes XXX" Line to the PR description to close it when merged